src/main/java/subway/Line.java

```diff
@@ -11,38 +14,69 @@ public class Line {
     private String name;
     @Column(length = 30, nullable = false)
     private String color;
-    @ManyToOne
-    @JoinColumn(name = "upward_station_id")
-    private Station upwardStation;
-    @ManyToOne
-    @JoinColumn(name = "downward_station_id")
-    private Station downwardStation;
-    @Column(nullable = false)
-    private int distance;
+    @OneToMany(cascade = CascadeType.ALL)
+    private List<Section> sections = new ArrayList<>();
 
     public Line() {
     }
 
-    public Line(Long id, String name, String color, Station upwardStation, Station downwardStation, int distance) {
-        this.id = id;
+    public Line(String name, String color, Section section) {
         this.name = name;
         this.color = color;
-        this.upwardStation = upwardStation;
-        this.downwardStation = downwardStation;
-        this.distance = distance;
+        sections.add(section);
     }
 
-    public Line(String name, String color, Station upwardStation, Station downwardStation, int distance) {
+    public void update(String name, String color) {
         this.name = name;
         this.color = color;
-        this.upwardStation = upwardStation;
-        this.downwardStation = downwardStation;
-        this.distance = distance;
     }
 
-    public void update(String name, String color) {
-        this.name = name;
-        this.color = color;
+    public void addSection(Section section) {
+        if (!isUpstationAddable(section.getUpwardStation())) {
+            throw new IllegalArgumentException("등록할 구간의 상행역은 해당 노선에 등록되어있는 하행 종점역이어야 합니다.");
+        }
+
+        if (!isdownStationAddable(section.getDownwardStation())) {
+            throw new IllegalArgumentException("이미 해당 노선에 등록되어있는 역은 새로운 구간의 하행역이 될 수 없습니다.");
+        }
+
+        sections.add(section);
+    }
+
+    private boolean isUpstationAddable(Station upStation) {
+        Section section = sections.get(0);
+        return section.getDownwardStation().equals(upStation);
+    }
+
+    private boolean isdownStationAddable(Station downStation) {
+        List<Station> stations = sections.stream()
+                .flatMap(section -> Stream.of(section.getUpwardStation(), section.getDownwardStation()))
+                .collect(Collectors.toList());
+
+        return !stations.contains(downStation);
+    }
+
+    public void deleteSection(Long stationId) {
+        if (sections.size() == 1) {
+            throw new NoSuchElementException("구간이 1개인 경우 구간을 삭제할 수 없습니다.");
+        }
+
+        Section section = sections.stream()
+                        .filter(element -> Objects.equals(element.getDownwardStation().getId(), stationId))
+                                .findFirst()
+                                        .orElseThrow(() -> new IllegalArgumentException(""));
+
+
+        if (!isSectionLastElement(section)) {
+            throw new IllegalArgumentException("마지막 구간에 해당하는 역만 삭제할 수 있습니다.");
+        }
+
+        sections.remove(sections.size() - 1);
```

Comment:
```suggestion
        sections.remove(section);
```

section으로도 삭제가 가능해 보이네요 :)